### ISSUE TITLE
What about DPS?

### ISSUE THREAD

**Issue by pingec** [justinleewells/pogo-optimizer#48]
What do you think about adding an extra column called DPS which would be calculated as Move.Power/Move.Duration

So when you have 2 pokemons of same type and with same Perfect score, you can judge which one to keep based on its DPS?

**Comment by justinleewells** [comment]
I intend to implement something like this over the weekend. Definitely a priority.

**Comment by Shauneh** [comment]
This is a good idea, there are plenty of resources available for this as well!

**Comment by jnsnow** [comment]
DPS for moves has been implemented, but I'd like to think about DPS that respects an attacker's ATK stat and considers STAB bonuses (and other modifiers.)

Looks like there's no real consensus yet on exactly what the damage formula looks like, it's all being back-correlated from repeated experiments.

Maybe we could compute a simple potential, though? Something like

ATK * DAM * STAB

would probably be sufficient for comparison purposes, and

((ATK * DAM * STAB) / move.DurationMs) * 1000

would be a reasonably sufficient "actual" DPS.... of course, it's just made up numbers... so we couldn't really call it DPS. Maybe "Damage Potential" is suitably vague.

If anyone can help point to a confirmed reverse engineering of the damage formula, we'll be able to use that to compute an accurate DPS.

**Comment by justinleewells** [comment]
Agreed ^. I'm going to add it when I add customizable columns because it would be pretty cramped right now.

**Comment by justinleewells** [comment]
Although I think the actual formula should probably be

ATK * ((DMG / DURATION) * STAB) 

I'm not sure how attack IVs factor into it though. Is it just ATKB + ATKIV = ATK?

**Comment by jnsnow** [comment]
ATK is ((BA + IVA) * CPM)
DEF is ((BD + IVD) * CPM)
(^ We should add these to the spreadsheet, btw.)
I'll use "POW" to refer to the strength of a given attack.

and the Pokemon Damage Formula generally works like:
http://bulbapedia.bulbagarden.net/wiki/Damage#Damage_formula

So in the case of PoGo, which people appear to believe follows a modified form of this formula, each attack follows something like:

(α * atk/def * pow + β) * STAB * [other modifiers]

Nobody seems to know what the alpha coefficient or the beta constant are, and we won't bother computing the opponent's DEF to compute a general "damage potential," so...

The best we can do is a guesstimation of the rough, raw damage we can cause:

ATK * POW * STAB

which just simply ignores the alpha and beta constants. It might be a good estimate of attacking potential. I don't think anyone has truly solved the damage equation yet otherwise...

**Comment by justinleewells** [comment]
Yeah, that sounds good to me. I'll try to add all of this into the next update. There are 2-3 other features that I'd like to get finished tonight as well.

**Comment by jnsnow** [comment]
It's fine, since the formula isn't known yet I'm not really pushing for a specific approximation. Take your time.
